File: Command/python-source/cookery/Bartender.py
from __future__ import annotations

from typing import ClassVar


class Bartender:
    """Второй исполнитель. Тоже ничего не знает про команды."""

    RECIPES: ClassVar[dict[str, list[str]]] = {
        "Mojito": ["Crush mint", "Add lime", "Add rum", "Add soda"],
        "Americano": ["Pour espresso", "Add water"],
    }

    def mix(self, drink: str) -> None:
        steps = self.RECIPES.get(drink)
        if steps is None:
            print(f"Bartender does not know how to make {drink}")
            return

        for step in steps:
            print(f"  {step}")
        print(f"  {drink} is ready")

File: JavaSource/src/main/java/ru/akhcheck/patterns/command/cookery/Bartender.java
package ru.akhcheck.patterns.command.cookery;

import java.util.List;
import java.util.Map;

/** Второй исполнитель. Тоже ничего не знает про команды. */
public class Bartender {
    private static final Map<String, List<String>> RECIPES = Map.of(
            "Mojito", List.of("Crush mint", "Add lime", "Add rum", "Add soda"),
            "Americano", List.of("Pour espresso", "Add water")
    );

    public void mix(String drink) {
        List<String> steps = RECIPES.get(drink);
        if (steps == null) {
            System.out.println("Bartender does not know how to make " + drink);
            return;
        }

        for (String step : steps) {
            System.out.println("  " + step);
        }
        System.out.println("  " + drink + " is ready");
    }
}
